src/System.Drawing.Common/src/System/Drawing/Font.cs

```diff
@@ -0,0 +1,586 @@
+            // have access to private/internal fields.
+            return font.FontFamily.Equals(FontFamily) &&
+                font.GdiVerticalFont == GdiVerticalFont &&
+                font.GdiCharSet == GdiCharSet &&
+                font.Style == Style &&
+                font.Size == Size &&
+                font.Unit == Unit;
+        }
+
+        /// <summary>
+        /// Gets the hash code for this <see cref='Font'/>.
+        /// </summary>
+        public override int GetHashCode()
+        {
+            return unchecked((int)((((uint)_fontStyle << 13) | ((uint)_fontStyle >> 19)) ^
+                         (((uint)_fontUnit << 26) | ((uint)_fontUnit >> 6)) ^
+                         (((uint)_fontSize << 7) | ((uint)_fontSize >> 25))));
+        }
+
+        /// <summary>
+        /// Returns a human-readable string representation of this <see cref='Font'/>.
+        /// </summary>
+        public override string ToString()
+        {
+            return string.Format(CultureInfo.CurrentCulture, "[{0}: Name={1}, Size={2}, Units={3}, GdiCharSet={4}, GdiVerticalFont={5}]",
+                                    GetType().Name,
+                                    FontFamily.Name,
+                                    _fontSize,
+                                    (int)_fontUnit,
+                                    _gdiCharSet,
+                                    _gdiVerticalFont);
+        }
+
+
+        /// <summary>
+        /// Returns the height of this Font in the specified graphics context.
+        /// </summary>
+        public float GetHeight(Graphics graphics)
+        {
+            if (graphics == null)
+            {
+                throw new ArgumentNullException(nameof(graphics));
+            }
+
+            float height;
+            int status = SafeNativeMethods.Gdip.GdipGetFontHeight(new HandleRef(this, NativeFont), new HandleRef(graphics, graphics.NativeGraphics), out height);
+            SafeNativeMethods.Gdip.CheckStatus(status);
+
+            return height;
+        }
+
+        public float GetHeight()
+        {
+            using (var screen = new Graphics.ScreenGraphics())
+            {
+                return GetHeight(screen.Graphics);
+            }
+        }
+
+        public float GetHeight(float dpi)
+        {
+            float height;
+            int status = SafeNativeMethods.Gdip.GdipGetFontHeightGivenDPI(new HandleRef(this, NativeFont), dpi, out height);
+            SafeNativeMethods.Gdip.CheckStatus(status);
+
+            return height;
+        }
+
+        /// <summary>
+        /// Gets the size, in points, of this <see cref='Font'/>.
+        /// </summary>
+        [Browsable(false)]
+        public float SizeInPoints
+        {
+            get
+            {
+                if (Unit == GraphicsUnit.Point)
+                {
+                    return Size;
+                }
+
+                using (var screen = new Graphics.ScreenGraphics())
```

Comment:
In all of these places where this ScreenGraphics helper is being used, aren't we incurring an extra allocation we weren't previously incurring?

Reply:
Presumably we can turn this helper into a a struct.

Reply:
Done